web/src/components/people/AdminView.tsx

```diff
@@ -39,42 +42,91 @@ export default function AdminView() {
     fetchPeople();
   }, []);
 
+  const updatePersonUI = (id: string, training: TrainingRecord) => {
+    // get people object and find the person with the right id
+    const personToUpdate = people!.find((person) => person.user.id === id);
+
+    if (personToUpdate && training) {
+      personToUpdate.training_record ??= [];
+      personToUpdate.training_record.push(training);
+    }
+  };
+
   if (!people) return null;
```

Comment:
`if (!people) return null;`

This would be better as loading state while `people` are being fetched.

Reply:
I think impossible as there needs to be at least the user accessing that page in the db